### PR TITLE
Adding Player::getDeviceOS() function

### PR DESCRIPTION
**Pull request by iiFlamiinBlaze**
## Introduction
This PR adds a new function to the player class, Player::getDeviceOS().

### API changes
 * Adds Player::getDeviceOS()

## Tests
Tested with a clean PMMP server and a test plugin, that sent a message when the player joins with the deviceOS number, to test if the deviceOS function worked.

### CODE REVIEW

File: src/pocketmine/Player.php [hunk 1]
```diff
@@ -3960,4 +3965,8 @@ public function getLoaderId() : int{
 	public function isLoaderActive() : bool{
 		return $this->isConnected();
 	}
+
+	public function getDeviceOS() : int{
+	    return $this->deviceOS;
```
Comment (supermaxalex) [inline src/pocketmine/Player.php]: formatting

File: src/pocketmine/Player.php [hunk 2]
```diff
@@ -234,6 +234,9 @@ public static function isValidUserName(?string $name) : bool{
 	/** @var string */
 	protected $xuid = "";
 
+	/** @var int */
```
Comment (supermaxalex) [inline src/pocketmine/Player.php]: Use `$deviceOs` instead of `$deviceOS`

**Review by moskadev — CHANGES_REQUESTED** [review]

**Review by dktapps — COMMENTED** [review]
I'm hesitant to merge this because it's going to conflict with changes I've made on `network-nuke`. I'm working on things like this on that branch (making login data more accessible to plugins).

I don't think this is particularly useful on its own, especially not without constants for the OS types. Without constants it's simply a magic number.

File: src/pocketmine/Player.php (hunk 2 above)
Comment (dktapps) [reply src/pocketmine/Player.php]: In my opinion this isn't necessary. `deviceOs` looks odd and unintuitive imho.

Comment (supermaxalex) [reply src/pocketmine/Player.php]: But it better fits how variables are called in PMMP.

File: src/pocketmine/Player.php (hunk 1 above)
Comment (iiFlamiinBlaze) [reply src/pocketmine/Player.php]: The formatting is fine...

File: src/pocketmine/Player.php (hunk 2 above)
Comment (iiFlamiinBlaze) [reply src/pocketmine/Player.php]: `$deviceOS` is better since the function is named `getDeviceOS()`

File: src/pocketmine/Player.php (hunk 1 above)
Comment (dktapps) [reply src/pocketmine/Player.php]: no, it's not. this line and the line below have incorrect indentation. It should use tabs, not spaces.

**Comment by MisteFr** [comment]
It might be a good idea to implement all the device stuff in a same PR.
You can get the device model too in the clientData part of the LoginPacket (clientData["DeviceModel"]).

**Pull request closed without merging**